content/articles/drug-labels-extraction-using-deep-learning/index.md

```diff
@@ -0,0 +1,212 @@
+
+### Extracting prescription medication labels using PaddleOCR
+
+This step involves visualizing the results. The `draw_ocr` method, `opencv` and `matplotlib` library help us achieve this.
+
+Let's extract the boxes, scores, and text coordinates into separate variables. We then pass these coordinates into the `draw_ocr` method. This is similar to object detection if you've ever performed a similar experiment. The `boxes` variable loops through and grabs the first index inside our `result` variable, the `texts` variable grabs the second, and the `scores` grab the third.
+
+```python
+boxes = [res[0] for res in result] 
+texts = [res[1][0] for res in result]
+scores = [res[1][1] for res in result]
+```
+The next step involves specifying the font path for our `draw_ocr` method. This is the reason why I mentioned earlier in the tutorial that you need to clone the PaddleOCR Github repository. I hope you do remember. To use the `draw_ocr` method, we need a path defined to the specific fonts. Inside the downloaded repo, when you follow that path shown below, you'll find different fonts in a folder. For our case, we use `latin.ttf`. `ttf` is the font format. It stands for `True Type Font`.
+
+We do this by issuing the following command:
+
+```python
+font_path = os.path.join('PaddleOCR', 'doc', 'fonts', 'latin.ttf')
+```
+Let's now load our image using OpenCV. We use OpenCV's `imread()` method to read in our image.
+
+```python
+img = cv2.imread(img_path) 
+img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+```
+
+The command `cv2.COLOR_BGR2RGB` reorders the color channels. By default, OpenCV imports images as BGR.
+
+The final step involves visualizing our images and detections. 
+
+Using matplotlib, let's first resize our display area.
+
+```python
+plt.figure(figsize=(15,15))
+```
+We then use the `draw_ocr` method to draw annotations on the image
+
+```python
+annotated = draw_ocr(img, boxes, texts, scores, font_path=font_path) 
+```
+
+Finally, we use matplotlib's `imshow()` method to visualize the image
+
+```python
+plt.imshow(annotated)
+```
+Let's take a look at the result. This's the same image we had earlier. We can see that it has extracted the text and drawn bounding boxes around it.
+
+![Output image using PaddleOCR](/engineering-education/drug-labels-extraction-using-deep-learning/output.png)
+
+Please find the full code to my code implementation [here](https://colab.research.google.com/drive/11EfSPacknXm16wWIkbyCCbyFSqVou0Ht).
+
+### Wrapping up
+One key feature of the PaddleOCR model is that it's extremely fast. We also note that with just a few lines of code, you can extract labels from prescribed medicine. Amazing, right? 
+
+In a nutshell, this tutorial shows you how you can use OCR to extract text from drug labels. The sizes and quality of your image will greatly impact your results. Try playing around with the code and see what you find.  
```

Comment:
Alert readers that the colab version may give an error since the image file may not be available to them remotely. Or advise them to download them then use their image paths for the task. Or advise them to carry out the task locally for the best outcome ie. faster processing due to local CPU/GPU usage, ease of adding images, etc.